src/pages/index.js

```diff
@@ -19,64 +21,192 @@ import {
   addNewCardButton,
   settings,
 } from "../utils/constants.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/API.js";
+import { data } from "autoprefixer";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "50644799-43da-42c8-ae19-a67fa2a948b2",
+    "Content-Type": "application/json",
+  },
+});
 
 const previewImagePopup = new PopupWithImage({
   popupSelector: "#card-preview-modal",
 });
-const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
-  const cardData = {
-    name: data.title,
-    link: data.url,
-  };
-  const newCard = createCard(cardData);
-  cardList.addItem(newCard);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-});
 
 const cardList = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (data) => {
       cardList.addItem(createCard(data));
     },
   },
   ".cards__list"
 );
 
+const confirmDeleteModal = new PopupWithConfirm(
+  "#delete-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm("#edit-avatar-modal", (input) => {
+  editAvatarModal.setLoading("Saving...");
+  handleAvatarEditSubmit(input);
+});
+editAvatarModal.setEventListeners();
+
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  const cardData = {
+    name: data.title,
+    link: data.url,
+  };
+
+  addCardPopup.setLoading("Saving...");
+
+  api
+    .addCards(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
```

Comment:
`resetValidation` is not needed here. 


`resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `disableButton` there.


```js

addFormValidator.disableButton();
```